_posts/2023-04-27-segment-replication.md

```diff
@@ -0,0 +1,425 @@
+        <td rowspan="4">Memory usage (%)</td>
+        <td class="border-rt">p50</td>
+        <td>35</td>
+        <td>23</td>
+        <td class="border-rt green-clr">&minus;34.29%</td>
+        <td>42</td>
+        <td>40</td>
+        <td class="green-clr">&minus;4.76%</td>
+    </tr>
+    <tr>
+        <td class="border-rt">p90</td>
+        <td>59</td>
+        <td>57</td>
+        <td class="border-rt green-clr">&minus;3.39%</td>
+        <td>59</td>
+        <td>63</td>
+        <td class="red-clr">6.78%</td>
+    </tr>
+    <tr>
+        <td class="border-rt">p99</td>
+        <td>69</td>
+        <td>61</td>
+        <td class="border-rt green-clr">&minus;11.59%</td>
+        <td>66</td>
+        <td>70</td>
+        <td class="red-clr">6.06%</td>
+    </tr>
+    <tr>
+        <td class="border-rt">p100</td>
+        <td>72</td>
+        <td>62</td>
+        <td class="border-rt green-clr">&minus;13.89%</td>
+        <td>69</td>
+        <td>72</td>
+        <td class="red-clr">4.35%</td>
+    </tr>
+    <tr>
+        <td>Error rate</td>
+        <td class="border-rt"></td>
+        <td>0.00%</td>
+        <td>0.00%</td>
+        <td class="border-rt">0.00%</td>
+        <td>0.00%</td>
+        <td>2.30%</td>
+        <td>2.30%</td>
+    </tr>
+</table>
+
+As the number of replicas increases, the amount of time required for primary shards to keep replicas up to date (known as the _replication lag_) also increases. This is because segment replication copies the segment files directly from primary shards to replicas.
+
+The benchmarking results show a non-zero error rate as the number of replicas increases. The error rate indicates that the [segment replication backpressure](#segment-replication-backpressure) mechanism is initiated when replicas cannot keep up with the primary shard. However, the error rate is offset by the significant CPU and memory gains that segment replication provides.
+
+## Other considerations
+
+The following considerations apply to segment replication in the 2.7 release:
+
+- **Read-after-write guarantees:**  The `wait_until` refresh policy is not compatible with segment replication.  If you use the `wait_until` refresh policy while ingesting documents, you'll get a response only after the primary node has refreshed and made those documents searchable.  Replica shards will respond only after having written to their local translog.  We are exploring other mechanisms for providing read-after-write guarantees. For more information, see the corresponding [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/6046).  
+
+- **System indexes** will continue to use document replication internally until read-after-write guarantees are available. In this case, document replication does not hinder the overall performance because there are few system indexes. 
+
+- [Enabling segment replication for an existing index](https://github.com/opensearch-project/OpenSearch/issues/3685) requires **reindexing**.
+
```

Comment:
- Rolling upgrades are also not yet supported. Upgrading to new versions of OpenSearch require a full cluster restart.